src/main/java/de/rwth/idsg/steve/config/BeanConfiguration.java

```diff
@@ -126,9 +125,7 @@ private void initDataSource() {
      * - http://stackoverflow.com/questions/32848865/jooq-dslcontext-correct-autowiring-with-spring
      */
     @Bean
-    public DSLContext dslContext() {
-        initDataSource();
-
+    public DSLContext dslContext(DataSource dataSource) {
```

Comment:
You can avoid the parameter if you get bean by calling `dataSource()` like in many spring samples.